src/main/java/eu/yals/ui/dev/AppInfoView.java

```diff
@@ -0,0 +1,114 @@
+
+@SpringComponent
+@UIScope
+@Route(value = Endpoint.UI.APP_INFO_PAGE, layout = AppView.class)
+@Caption("App Info")
+@Icon(VaadinIcon.INFO)
+@PageTitle("Link shortener for friends: App Info")
+public class AppInfoView extends VerticalLayout {
+    private static final String UNDEFINED = "UNDEFINED";
+
+    private final GitService gitService;
+    private final GitRepoState gitRepoState;
+    private final MavenInfo mavenInfo;
+    private final AppUtils appUtils;
+
+    public AppInfoView(GitService gitService, GitRepoState gitRepoState, MavenInfo mavenInfo, AppUtils appUtils) {
+        this.gitService = gitService;
+        this.gitRepoState = gitRepoState;
+        this.mavenInfo = mavenInfo;
+        this.appUtils = appUtils;
+
+        init();
+    }
+
+    private void init() {
+        setId(ID.VIEW_ID);
+
+        HorizontalLayout publicInfoArea = publicInfoArea();
+        add(publicInfoArea);
+
+        if (appUtils.isDevelopmentModeActivated() || appUtils.hasDevHeader()) {
+            VerticalLayout devInfoArea = devInfoArea();
+            add(devInfoArea);
+        }
+    }
+
+    private HorizontalLayout publicInfoArea() {
+        HorizontalLayout publicArea = new HorizontalLayout();
+        publicArea.setId(ID.PUBLIC_INFO_AREA);
+
+        String latestTag = gitService.getLatestTag();
+        String latestCommit = gitService.getLatestCommit();
+
+        Span versionStart = new Span(String.format("Version %s (based on commit ", latestTag));
+        Anchor commit =
+                new Anchor(
+                        String.format("%s/%s", App.Git.REPOSITORY, latestCommit),
+                        latestCommit.substring(0, Integer.min(latestCommit.length(), 7)));
+        commit.setId(ID.COMMIT_LINK);
+        Span versionEnd = new Span(")");
+
+        Span version = new Span(versionStart, commit, versionEnd);
+        version.setId(ID.VERSION);
+
+        publicArea.setWidthFull();
+
+        publicArea.add(version);
+
+        return publicArea;
+    }
+
+    private VerticalLayout devInfoArea() {
+        VerticalLayout devInfoArea = new VerticalLayout();
+        devInfoArea.setId(ID.DEV_INFO_AREA);
+
+        String vaadinVersionStr = mavenInfo.hasValues() ? mavenInfo.getVaadinVersion() : UNDEFINED;
+        String vaadinFlowVersion = Version.getFullVersion();
+
+        String gitBranchStr = gitRepoState.hasValues() ? gitRepoState.getBranch() : UNDEFINED;
+        String gitHostStr = gitRepoState.hasValues() ? gitRepoState.getBuildHost() : UNDEFINED;
+
+        H3 h3 = new H3("Dev Info");
+        Span vaadinVersion = new Span("Vaadin version: " + vaadinVersionStr +
+                " (Flow: " + vaadinFlowVersion + ")");
+
+        Span gitBranch = new Span("Git branch: " + gitBranchStr);
+        Span gitHost = new Span("Built at " + gitHostStr);
+
+        devInfoArea.add(h3, vaadinVersion, gitBranch, gitHost);
+
+        add(devInfoArea);
+        return devInfoArea;
+    }
+
+    public static class ID {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'ID' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5385406)